src/web/mdns-publisher/mdns_publisher.cpp

```
@@ -28,267 +28,277 @@
+    if (avahi_entry_group_is_empty(mClientGroup))
+    {
+        syslog(LOG_ERR, "Adding service '%s'", mServiceName);
 
-    VerifyOrExit(avahi_server_add_service(aServer, mGroup, AVAHI_IF_UNSPEC,
-                                          AVAHI_PROTO_UNSPEC, (AvahiPublishFlags) 0,
-                                          mServiceName, mType, NULL, NULL, mPort,
-                                          mNetworkNameTxt, mExtPanIdTxt, NULL) == 0,
-                 ret = kMdnsPublisher_FailedAddSevice);
+        VerifyOrExit(avahi_entry_group_add_service(mClientGroup, AVAHI_IF_UNSPEC,
+                                                   AVAHI_PROTO_UNSPEC, (AvahiPublishFlags) 0,
+                                                   mServiceName, mType, NULL, NULL, mPort,
+                                                   mNetworkNameTxt, mExtPanIdTxt, NULL) == 0,
+                     ret = kMdnsPublisher_FailedAddSevice);
 
-    syslog(LOG_INFO, " Service Name: %s Port: %d Network Name: %s Extended Pan ID: %s ",
-           mServiceName, mPort, mNetworkNameTxt, mExtPanIdTxt);
+        syslog(LOG_INFO, " Service Name: %s  Port: %d  Network Name: %s  Extended Pan ID: %s ",
+               mServiceName, mPort, mNetworkNameTxt, mExtPanIdTxt);
 
-    VerifyOrExit(avahi_s_entry_group_commit(mGroup) == 0,
-                 ret = kMdnsPublisher_FailedRegisterSevice);
-    return ret;
+        VerifyOrExit(avahi_entry_group_commit(mClientGroup) == 0,
+                     ret = kMdnsPublisher_FailedRegisterSevice);
+    }
 
 exit:
-    if (ret != kMdnsPublisher_OK)
+
+    if (ret == kMdnsPublisher_FailedAddSevice)
+    {
+        serviceName = avahi_alternative_service_name(mServiceName);
+        avahi_free(mServiceName);
+        mServiceName = serviceName;
+        syslog(LOG_INFO, "Service name collision, renaming service to '%s'", mServiceName);
+        avahi_entry_group_reset(mClientGroup);
+        CreateService(aClient);
```

Is it intended to ignore error from `CreateService()`?